File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] inter) {
        int n=inter.length;
        List<List<Integer>>list=new ArrayList<>();
        Arrays.sort(inter, (a, b) -> Integer.compare(a[0], b[0]));
        list.add(new ArrayList<>(Arrays.asList(inter[0][0], inter[0][1])));
        for(int i=1;i<n;i++){
            List<Integer>li=list.get(list.size()-1);
            int val=li.get(1);
            if(inter[i][0]<=val){
                list.get(list.size()-1).set(1,Math.max(val,inter[i][1]));
            }else{
                list.add(new ArrayList<>(Arrays.asList(inter[i][0],inter[i][1])));
            }
        }
        int [][]ans=new int [list.size()][2];
        for(int j=0;j<list.size();j++){
            for(int i=0;i<2;i++){
                ans[j][i]=list.get(j).get(i);
            }
        }
        return ans;
    }
}

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& i) {
       sort(i.begin(),i.end(),[](const auto & a,const auto & b){
            return a[0]<b[0];
       });
       vector<vector<int>>vec;
       int n=i.size();
       vec.push_back({i[0][0],i[0][1]});
       for(int j=1;j<n;j++){
        if(i[j][0]<=vec.back()[1]){
            vec.back()[1]=max(vec.back()[1],i[j][1]);
        }else{
            vec.push_back({i[j][0],i[j][1]});
        }
       }
       return vec;
    }
};